src/radical/pilot/agent/agent_0.py

```diff
@@ -351,29 +354,36 @@ def _start_sub_agents(self):
                 #        out for the moment, which will make this unable to
                 #        work with a number of launch methods.  Can the
                 #        offset computation be moved to the LRMS?
-                ls_name = "%s/%s.sh" % (os.getcwd(), sa)
-                slots   = copy.deepcopy(self._cfg['lrms_info'])
-                slots['cpu_processes'] = 1
-                slots['cpu_threads'  ] = 1
-                slots['gpu_processes'] = 0
-                slots['gpu_threads'  ] = 0
-                slots['nodes'        ] = [{'name'    : node[0],
-                                           'uid'     : node[1],
-                                           'core_map': [[0]],
-                                           'gpu_map' : [],
-                                           'lfs'     : {'path': '/tmp', 'size': 0}
-                                          }]
+                bs_name = "%s/bootstrap_2.sh" % (self._pwd, sa)
+                ls_name = "%s/%s.sh" % (self._pwd, sa)
+                slots = {
```

Comment:
Here you give a slot to the agent, right?

Reply:
Indeed, in case a subagent needs to be placed on a compute node.  The RM will make sure that the respective node is blocked from the scheduler.